fontforge/fvfonts.c

```diff
@@ -1256,8 +1254,11 @@ return;
     /*  If both are CID then merge each subfont separately */
     if ( fv->sf->cidmaster!=NULL && other->subfonts!=NULL )
 	CIDMergeFont(fv->sf->cidmaster,other,preserveCrossFontKerning);
-    else
+    else {
+	if (other->subfonts!=NULL)
```

Comment:
Here we flatten if there are subfonts as the comment on (old) line 1254 suggests. 

I take it that the loop level I removed was intended do be equivalent to what `SFFlatten()` does, and perhaps faster or better in some other way. I discuss in the comments on #3347 why the code doesn't work and would be difficult to make work.